src/parquet/file/metadata.h

```diff
@@ -91,6 +91,32 @@ class PARQUET_EXPORT RowGroupMetaData {
 
 class FileMetaDataBuilder;
 
+struct PARQUET_EXPORT FileVersion {
+  /// Application that wrote the file. e.g. "IMPALA"
+  std::string application;
+
+  /// Version of the application that wrote the file, expressed in three parts
+  /// (<major>.<minor>.<patch>). Unspecified parts default to 0, and extra parts are
+  /// ignored. e.g.:
+  /// "1.2.3"    => {1, 2, 3}
+  /// "1.2"      => {1, 2, 0}
+  /// "1.2-cdh5" => {1, 2, 0}
+  struct {
+    int major;
+    int minor;
+    int patch;
+  } version;
+
+  FileVersion() {}
+  explicit FileVersion(const std::string& created_by);
+
+  /// Returns true if version is strictly less than <major>.<minor>.<patch>
+  bool VersionLt(int major, int minor = 0, int patch = 0) const;
```

Comment:
I mean, we could combine `bool VersionEq` and `bool VersionLt` into `int compare(FileVersion& fv)`. This will require modifying `VersionLt(1, 2, 9)` to `compare(FileVersion(1, 2, 9)) < 0`.
 I also prefer `FileVersion` as the argument since we can extend `struct version` and still use the same `compare or VersionEq + VersionLt`.
PARQUET-793 will require extending `struct version`.